subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildArtifactTransformIntegrationTest.groovy

```diff
@@ -48,10 +57,29 @@ class CompositeBuildArtifactTransformIntegrationTest extends AbstractCompositeBu
                     artifactTransform(XForm)
                 }
             }
+            
+            task resolve {
+                def artifacts = configurations.compileClasspath.incoming.artifactView { 
+                    attributes.attribute(Attribute.of("artifactType", String), "xform")
+                }.artifacts
+                inputs.files artifacts.artifactFiles
+                doLast {
+                    artifacts.each { println it }
+                }
+            }
         """
         expect:
-        execute(buildA, "assemble")
+        execute(buildA, "resolve")
         assertTaskExecuted(":buildB", ":jar")
         assertTaskExecuted(":buildC", ":jar")
+
+        outputContains("buildB-1.0.jar (project :buildB)")
+        outputContains("buildC-1.0.jar (project :buildC)")
+        output.count("Transforming") == 2
+
+        if (IncrementalArtifactTransformationsRunner.incrementalArtifactTransformations) {
+            outputContains(buildB.file("build/transforms/buildB-1.0.jar").absolutePath)
+            outputContains(buildC.file("build/transforms/buildC-1.0.jar").absolutePath)
```

Comment:
It's hard to figure out what this check belongs to. We should not just print the path, but add some context around it, like `println "Resolved artifact: $it"`.

Reply:
Done.

Reply:
It's a bit better, but what I had in mind was to have a specific `println` for this guy _in the test task,_ and then check that string in the end. We are now still testing the standard `Transforming artifact...` string instead.

Anyway, I don't feel very strongly anymore, so it's fine like this, too.